docs/synapse/kubernetes/optic.yaml

```diff
@@ -0,0 +1,122 @@
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: example-optic00
+  labels:
+    app.kubernetes.io/name: "optic"
+    app.kubernetes.io/instance: "optic00"
+    app.kubernetes.io/version: "v2.x.x"
+    app.kubernetes.io/component: "optic"
+    app.kubernetes.io/part-of: "synapse"
+    environment: "dev"
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 10Gi
+  # You will need to use an appropriate storageClassName for your cluster.
+  storageClassName: do-block-storage
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: optic00
+  labels:
+    app.kubernetes.io/name: "optic"
+    app.kubernetes.io/instance: "optic00"
+    app.kubernetes.io/version: "v2.x.x"
+    app.kubernetes.io/component: "optic"
+    app.kubernetes.io/part-of: "synapse"
+    environment: "dev"
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: "optic"
+      app.kubernetes.io/instance: "optic00"
+      app.kubernetes.io/version: "v2.x.x"
+      app.kubernetes.io/component: "optic"
+      app.kubernetes.io/part-of: "synapse"
+      environment: "dev"
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: "optic"
+        app.kubernetes.io/instance: "optic00"
+        app.kubernetes.io/version: "v2.x.x"
+        app.kubernetes.io/component: "optic"
+        app.kubernetes.io/part-of: "synapse"
+        environment: "dev"
+    spec:
+      securityContext:
+        runAsUser: 999
+        runAsGroup: 999
+        fsGroup: 999
+      volumes:
+        - name: data
+          persistentVolumeClaim:
+            claimName: example-optic00
+      containers:
+      - name: optic
+        image: vertexproject/optic:v2.x.x
+        securityContext:
+          readOnlyRootFilesystem: true
+        env:
+          - name: SYN_LOG_LEVEL
+            value: DEBUG
+          - name: SYN_LOG_STRUCT
+            value: "false"
+          - name: SYN_OPTIC_AHA_PROVISION
+            value: "ssl://aha.default.svc.cluster.local:27272/..."
```

Comment:
Is the trailing `...` here on purpose?

Reply:
We need to have the PROVISION envar available to for folks to drop in easily. I see a few options for the examples:

1. empty value - would fail to provision.
2. stub value ( which I have here ) - gives a visual anchor that can be found and replaced.
3. broken string encoding - that breaks the inline YAML highlighting by sphinx but forces a user who copy+pastes the data to fix it before it can be processed.

Reply:
I just didn't notice anywhere that this "replace me" convention was discussed :) Did i miss it?

Reply:
Should be here https://github.com/vertexproject/synapse/pull/3047/files#diff-ced6b121494d6653b3ecb07b1bcd79d65f8e6087864bff155f6238a8bd458518R1003-R1017 - each service has an associated block for that